### ISSUE TITLE
Report savings on all opportunities combined

### ISSUE THREAD

**Issue by rsedykh** [GoogleChrome/lighthouse#8740]
**TL;DR: I want to know your opinion about prediction model for Performance Score related to image optimization**

I wanted to figure out the *exact* impact of image optimizations on Performance Score before implementing them.

I took the world’s top 500 websites (from Alexa) and ran performance analyses with Lighthouse and PageDetox (a diagnostic tool made by Uploadcare, a company I work at).

PageDetox fetched website images, optimized them, and calculated the differences in image weights and page-loading times (I used the same throttle settings as Lighthouse, and decreased derived load times by `saved_bytes / speed`).

From there, I tried to model the improvement in Performance Score by proportionally decreasing the weighted params used in the score calculations: FCP, FMP, SI, CPU, and TTI.

I understand that I have the two weak links in this model:
1) Load time improvement might easily fall to zero if JS rendering takes ages while even the largest images have already been loaded;
2) Linearly decreasing the Performance Score parameters may not be an exact fit.

But I think that (on average) the data I got is legit. Here is a raw data — https://docs.google.com/spreadsheets/d/1K1qUIPCm2ZOxwCW52Zu5oLo5p9r8xydEh5h97HF3eyU/edit?usp=sharing

**What are your thoughts about this prediction model? Where do you see the room for improvement?**

My final goal is to get *proof* to convince people (by estimating the value they get *prior* to investing any resources in the optimization) that image optimization is essential and affects business metrics.

p.s. If you are interested, I published an article about this model — https://blog.uploadcare.com/image-optimization-and-performance-score-23516ebdd31d. Though I've mentioned above everything related to the model.

![](https://ucarecdn.com/fac257d3-a140-42cc-95bd-a651537fb65a/report_top500.png)

**Comment by patrickhulce** [comment]
This is *awesome* @rsedykh!!

I'd be really interested to compare our "estimated savings" time to the results of your data (I saw the score for our opportunities in the raw data, but couldn't find the estimated savings in milliseconds anywhere?).

Funnily enough, the "estimated savings" time in most of our opportunities is derived exactly from the estimated impact to the TTI metric. A lazy model to compare these results to would be to subtract that number from TTI, recompute the overall score, and report the difference.

Investigating any cases you've found where we don't line up very well with the real-world savings would be super helpful to improve our opportunities!

**Comment by rsedykh** [comment]
@patrickhulce thanks! Can you tell me exact JSON values of ms savings after image optimizations report that I should use? I saw are a bunch of savings, and they sometimes overlap (there were more savings than the actual image size, for example).

So then I just subtract these savings (in ms) from TTI, recompute TTI score, recompute Performance score — that's it? Shouldn't I modify the other four parameters in any way?

I'd also add a comparison of LH savings in ms vs PD savings in ms just to see if they are close, hope they will be similar. :-)

**Comment by patrickhulce** [comment]
Ah, yeah we don't output an overall savings that has all the opportunities and they can't be added up for the reasons you've already discovered. 

It's something to consider though!

**Comment by rsedykh** [comment]
@patrickhulce so basically in the future Lighthouse report might say you exactly how much you can save by implementing opportunities? Nice! Until then, people can use pagedetox.com. :-)

(Right now I'm considering adding Lighthouse report to the mix so users can see savings + increase of Performance Score, but it will be more user-friendly for users to actually see it within Lighthouse).

**Comment by paulirish** [comment]
We're not planning to calculate this number.
